generators/firechip/src/main/scala/FireSim.scala

```diff
@@ -38,44 +40,113 @@ object NodeIdx {
 /**
   * Under FireSim's current multiclock implementation there can be only a
   * single clock bridge. This requires, therefore, that it  be instantiated in
-  * the harness and reused across all supernode instances. This class attempts to 
+  * the harness and reused across all supernode instances. This class attempts to
   * memoize its instantiation such that it can be referenced from within a ClockScheme function.
   */
 class ClockBridgeInstantiator {
-  private var _clockRecord: Option[RecordMap[Clock]] = None
+  private var _harnessClockMap: HashMap[String, (Double, Clock)] = HashMap.empty
+
+  // Assumes that the supernode implementation results in duplicated clocks
+  //   (i.e. only 1 set of clocks is generated for all BuildTop designs)
+  private var _ratClockMap: HashMap[String, (RationalClock, Clock)] = HashMap.empty
```

Comment:
This can and should be a val since the data structure itself mutable. I think i'd also prefer if you used a `mutable.LinkedHashMap` just to future proof it against someone iterating on it to generate HW.

Reply:
Same for the structure above.

Reply:
Also these two structures are analogs for harness and chip clocks right? If so, can you make the names reflect that by renaming _ratClockMap to _chipClockMap or something?

Reply:
Yup. I can rename things better.